File: backend/app/schemas/order.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from decimal import Decimal


class OrderItemCreate(BaseModel):
    product_id: int
    quantity: int


class OrderCreate(BaseModel):
    items: List[OrderItemCreate]
    shipping_address: str
    recipient_name: Optional[str] = None
    recipient_phone: Optional[str] = None
    payment_method: str = "cod"
    voucher_code: Optional[str] = None
    note: Optional[str] = None


class OrderItemResponse(BaseModel):
    order_item_id: int
    product_id: int
    product_name: Optional[str]
    product_image: Optional[str]
    quantity: int
    price_at_order: Decimal  # Đổi từ str → Decimal (khớp Prisma)

    class Config:
        from_attributes = True


class OrderResponse(BaseModel):
    order_id: int
    order_number: Optional[str]       # Prisma: orderNumber
    user_id: int
    shop_id: Optional[int]
    total_price: Decimal
    discount_amount: Optional[Decimal]  # Prisma: discountAmount
    final_price: Decimal
    shipping_fee: Optional[Decimal]   # Prisma: shippingFee (mới)
    payment_method: str
    payment_status: str
    order_status: str
    shipping_address: Optional[str]
    recipient_name: Optional[str]
    recipient_phone: Optional[str]
    notes: Optional[str]              # Prisma: notes
    voucher_id: Optional[int]
    created_at: Optional[datetime]
    items: List[OrderItemResponse] = []

    class Config:
        from_attributes = True


class OrderListResponse(BaseModel):
    orders: List[OrderResponse]
    total: int
    page: int
    pages: int


class OrderStatusUpdate(BaseModel):
    reason: Optional[str] = None


class TrackingResponse(BaseModel):
    order_id: int
    order_status: str
    shipment: Optional[dict] = None

File: frontend/src2/types/order.ts
export type OrderStatus = 'pending' | 'confirmed' | 'ready_to_ship' | 'shipped' | 'delivered' | 'completed' | 'cancelled' | 'returned'
export type PaymentMethod = 'momo' | 'cod' | 'vnpay'
export type PaymentStatus = 'unpaid' | 'paid' | 'failed' | 'refunded'

export interface OrderItem {
  order_item_id: number
  product_id: number
  product_name?: string
  product_image?: string
  quantity: number
  price_at_order: string
}

export interface ShipperInfo {
  shipper_id: number
  name: string
  phone?: string
  vehicle_type?: string
  license_plate?: string
  rating: string
}

export interface Shipment {
  shipment_id: number
  status: string
  shipper_id?: number
  pickup_location?: string
  delivery_location?: string
  current_location?: { lat: number; lng: number }
  pickup_time?: string
  delivery_time?: string
}

export interface Order {
  order_id: number
  order_number?: string
  user_id: number
  shop_id?: number
  shop_name?: string
  shop_address?: string
  shipper_id?: number
  shipper_info?: ShipperInfo
  total_price: string
  discount: string
  final_price: string
  payment_method: PaymentMethod
  payment_status: PaymentStatus
  order_status: OrderStatus
  shipping_address: string
  recipient_name?: string
  recipient_phone?: string
  note?: string
  created_at?: string
  items: OrderItem[]
  shipment?: Shipment
}

export interface CartItem {
  cart_id: number
  product_id: number
  product_name?: string
  product_image?: string
  price: string
  quantity: number
  shop_id?: number
  shop_name?: string
  stock_quantity?: number
}

export interface Cart {
  items: CartItem[]
  total: number
  item_count: number
}

export interface CheckoutData {
  items: { product_id: number; quantity: number }[]
  shipping_address: string
  recipient_name?: string
  recipient_phone?: string
  payment_method: PaymentMethod
  voucher_code?: string
  note?: string
}

